pokedex/share.lua

```diff
@@ -6,20 +6,21 @@ local monarch = require "monarch.monarch"
 local dex = require "pokedex.dex"
 local pokedex = require "pokedex.pokedex"
 local statuses = require "pokedex.statuses"
+local _file = require "utils.file"
+local platform = require "utils.platform"
 
 local M = {}
 
-local function load_json(j)
-	local json_data = nil
-	-- Use pcall to catch possible parse errors so that we can print out the name of the file that we failed to parse
-	if pcall(function() json_data = json.decode(j) end) then
-		return json_data
-	else
-		return nil
-	end
-end
+-- For checking if sharing is enabled
+M.ENABLED = {
+	CLIPBOARD = clipboard ~= nil,
+	QRCODE_GENERATE = true,
+	QRCODE_READ = platform.PHONE or platform.MACOS,
+}
+M.ENABLED.ANY = M.ENABLED.CLIPBOARD or M.ENABLED.QRCODE_READ
```

Comment:
Not super happy with this bit, want a single place to define when a sharing functionality is enabled in case there are updates to the library we use. Guess it's not pretty but gets the job done